File: frontend/src/microservices/operations/objects/ManagerBatchDecisionResponse.ts
// 本文件定义 operations 模块的 `ManagerBatchDecisionResponse`，作为响应数据并提供 JSON 编解码。

export type ManagerBatchDecisionResponse = {
  processedCount: number
  orderItemIds: string[]
  action: string
}
export const managerBatchDecisionResponseFromJson = (json: string): ManagerBatchDecisionResponse =>
  JSON.parse(json) as ManagerBatchDecisionResponse

export const managerBatchDecisionResponseToJson = (value: ManagerBatchDecisionResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/overall/objects/ManagerBatchDecisionResponse.scala
// 本文件定义批量和单条管理动作返回的结果对象。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerBatchDecisionResponse(processedCount: Int, orderItemIds: List[String], action: String)
object ManagerBatchDecisionResponse:
  given sourceEncoder: Encoder[ManagerBatchDecisionResponse] = deriveEncoder
  given sourceDecoder: Decoder[ManagerBatchDecisionResponse] = deriveDecoder
